test/wasm.simd/replaceLaneTests.js

```diff
@@ -0,0 +1,67 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft Corporation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+let passed = true;
+
+function assertEquals(expected, actual) {
+    if (expected != actual) {
+        passed = false;
+        throw `Expected ${expected}, received ${actual}`;
+    }
+}
+
+const INITIAL_SIZE = 1;
+const memObj = new WebAssembly.Memory({initial:INITIAL_SIZE});
+const arrays = [new Int32Array (memObj.buffer),
+    new Int16Array (memObj.buffer),
+    new Int8Array (memObj.buffer),
+    new Float32Array (memObj.buffer)
+];
+
+const module = new WebAssembly.Module(readbuffer('replace.wasm'));
+const instance = new WebAssembly.Instance(module, { "dummy" : { "memory" : memObj } }).exports;
+
+let testIntFloatReplace = function (funcname, splat, val, len) {
+
+    let typeIndex = Math.log2(len) - 2;
+    let arr = arrays [typeIndex];
+
+    if (len == 32) { //special case for floats
```

Comment:
we could change the indexing to a dict w/ an array view per each type where the type comes from a test name if this version is too ugly :-)